tools/pubsys/src/kit/publish_kit/mod.rs

```diff
@@ -0,0 +1,182 @@
+use crate::Args;
+use buildsys_config::DockerArchitecture;
+use clap::Parser;
+use log::{debug, info, trace};
+use pubsys_config::InfraConfig;
+use regex::Regex;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::path::PathBuf;
+use tokio::process::Command;
+
+/// Takes a local kit built using buildsys and publishes it to a vendor specified in Infra.toml
+#[derive(Debug, Parser)]
+pub(crate) struct PublishKitArgs {
+    /// Location of the local kit
+    #[arg(long)]
+    kit_path: PathBuf,
+
+    /// Vendor to publish kit to
+    #[arg(long)]
+    vendor: String,
+
+    /// The version and build ID of the kit that should be published, e.g. v1.0.0-abcd123
+    #[arg(long)]
+    version: String,
+}
+
+macro_rules! docker {
+    ($arg: expr) => {{
+        let result = Command::new("docker")
+            .args($arg)
+            .output()
+            .await
+            .context(error::CommandSnafu { command: "docker" })?;
+        ensure!(
+            result.status.success(),
+            error::PublishFailSnafu {
+                reason: String::from_utf8_lossy(&result.stderr)
+            }
+        );
+        result.stdout
+    }};
+}
+
+pub(crate) async fn run(args: &Args, publish_kit_args: &PublishKitArgs) -> Result<()> {
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let infra_config = InfraConfig::from_path_or_lock(&args.infra_config_path, false)
+        .context(error::ConfigSnafu)?;
+    trace!("Parsed infra config: {:?}", infra_config);
+
+    // Fetch the vendor container registry uri
+    let vendor = infra_config
+        .vendor
+        .as_ref()
+        .context(error::NoVendorsSnafu)?
+        .get(&publish_kit_args.vendor)
+        .context(error::VendorNotFoundSnafu {
+            name: publish_kit_args.vendor.clone(),
+        })?;
+    let vendor_registry_uri = vendor.registry.clone();
+    debug!(
+        "Found vendor container registry at uri: {}",
+        vendor_registry_uri
+    );
+
+    // Auto resolve the expected paths for the kit contents archive
+    let kit_path = publish_kit_args.kit_path.as_path();
+    let kit_name = kit_path
+        .file_name()
+        .context(error::InvalidPathSnafu { path: &kit_path })?
+        .to_string_lossy();
+    let kit_version = publish_kit_args.version.clone();
+
+    let mut platform_images = Vec::new();
+    for arch in ["aarch64", "x86_64"] {
+        let docker_arch =
+            DockerArchitecture::try_from(arch).context(error::InvalidArchitectureSnafu)?;
+
+        let kit_filename = format!("{}-{}-{}.tar", &kit_name, &kit_version, arch);
+        let path = kit_path.join(&kit_filename);
+
+        let out = docker!(["load", format!("--input={}", path.display()).as_str(),]);
```

Comment:
For ease of local development, I think we could skip over architectures where the kit is not available. But that creates its own headaches so this is good for now.